Packs/CyberArkPAS/Integrations/CyberArkPAS/CyberArkPAS_test.py

```diff
@@ -0,0 +1,192 @@
+
+
+@pytest.mark.parametrize('command, args, http_response, context', [
+
+   (add_user_command, ADD_USER_ARGS, ADD_USER_RAW_RESPONSE, ADD_USER_CONTEXT),
+   (update_user_command, UPDATE_USER_ARGS, UPDATE_USER_RAW_RESPONSE, UPDATE_USER_CONTEXT),
+   (get_users_command, {}, GET_USERS_RAW_RESPONSE, GET_USERS_CONTEXT),
+
+   (add_safe_command, ADD_SAFE_ARGS, ADD_SAFE_RAW_RESPONSE, ADD_SAFE_CONTEXT),
+   (update_safe_command, UPDATE_SAFE_ARGS, UPDATE_SAFE_RAW_RESPONSE, UPDATE_SAFE_CONTEXT),
+   (get_list_safes_command, {}, GET_LIST_SAFES_RAW_RESPONSE, GET_LIST_SAFES_CONTEXT),
+   (get_safe_by_name_command, GET_SAFE_BY_NAME_ARGS, GET_SAFE_BY_NAME_RAW_RESPONSE, GET_SAFE_BY_NAME_CONTEXT),
+
+   (add_safe_member_command, ADD_SAFE_MEMBER_ARGS, ADD_SAFE_MEMBER_RAW_RESPONSE, ADD_SAFE_MEMBER_CONTEXT),
+   (update_safe_member_command, UPDATE_SAFE_MEMBER_ARGS, UPDATE_SAFE_MEMBER_RAW_RESPONSE, UPDATE_SAFE_MEMBER_CONTEXT),
+   (list_safe_members_command, LIST_SAFE_MEMBER_ARGS, LIST_SAFE_MEMBER_RAW_RESPONSE, LIST_SAFE_MEMBER_CONTEXT),
+
+   (add_account_command, ADD_ACCOUNT_ARGS, ADD_ACCOUNT_RAW_RESPONSE, ADD_ACCOUNT_CONTEXT),
+   (update_account_command, UPDATE_ACCOUNT_ARGS, UPDATE_ACCOUNT_RAW_RESPONSE, UPDATE_ACCOUNT_CONTEXT),
+   (get_list_accounts_command, GET_LIST_ACCOUNT_ARGS, GET_LIST_ACCOUNT_RAW_RESPONSE, GET_LIST_ACCOUNT_CONTEXT),
+   (get_list_account_activity_command, GET_LIST_ACCOUNT_ACTIVITIES_ARGS, GET_LIST_ACCOUNT_ACTIVITIES_RAW_RESPONSE, GET_LIST_ACCOUNT_ACTIVITIES_CONTEXT),
+])
+def test_cyberark_aim_commands(command, args, http_response, context, mocker):
+    """Unit test
+    Given
+    - demisto args
+    - raw response of the http request
+    When
+    - mock the http request result
+    Then
+    - create the context
+    - validate the expected_result and the created context
+    """
+    mocker.patch.object(Client, '_generate_token')
+    client = Client(server_url="https://api.cyberark.com/", username="user1", password="12345", use_ssl=False,
+                    proxy=False)
+
+    mocker.patch.object(Client, '_http_request', return_value=http_response)
+
+    outputs = command(client, **args)
+    results = outputs.to_context()
+    assert results.get("EntryContext") == context
+
+
+def test_fetch_incidents(mocker):
+    """Unit test
+    Given
+    - raw response of the http request
+    When
+    - mock the http request result as 5 results that are sorted from the newest to the oldest
+    Then
+    - as defined in the demisto params - show only 2, those should be the oldest 2 available
+    - validate the incidents values
+    """
+    mocker.patch.object(Client, '_generate_token')
+    client = Client(server_url="https://api.cyberark.com/", username="user1", password="12345", use_ssl=False,
+                    proxy=False)
+
+    mocker.patch.object(Client, '_http_request', return_value=GET_SECURITY_EVENTS_RAW_RESPONSE)
+
+    a, incidents = fetch_incidents(client, {}, "3 days", "0", "2")
+    assert incidents == INCIDENTS
+
+
```

Comment:
please add 2 more test cases.
1. with a query/score which is bigger that some of the incidents and validate that only the relevant are being fetched.
1. with num of incidents > max_fetch and validate that not all of them were fetched.